core/src/miner/mem_pool.rs

```diff
@@ -865,7 +865,81 @@ impl MemPool {
             .count()
     }
 
-    /// Return all future transactions.
+    /// Return all future transactions along with current transactions.
+    pub fn get_future_pending_transactions(
+        &self,
+        size_limit: usize,
+        current_timestamp: Option<u64>,
+        range: Range<u64>,
+    ) -> PendingSignedTransactions {
+        let mut current_size: usize = 0;
+        let future_pending_items: Vec<_> = self
+            .future
+            .queue
+            .iter()
+            .map(|t| {
+                self.by_hash
+                    .get(&t.hash)
+                    .expect("All transactions in `current` and `future` are always included in `by_hash`")
+            })
+            .filter(|t| {
+                if let Some(expiration) = t.expiration() {
+                    if let Some(timestamp) = current_timestamp {
+                        return expiration >= timestamp
+                    }
+                }
+                true
+            })
+            .filter(|t| range.contains(&t.inserted_timestamp))
+            .take_while(|t| {
+                let encoded_byte_array = rlp::encode(&t.tx);
+                let size_in_byte = encoded_byte_array.len();
+                current_size += size_in_byte;
+                current_size < size_limit
+            })
+            .collect();
+
+
```

Comment:
Please remove a new line. We are not using two blank lines.
Also, please check your PR before request a review.
You can fix these things before requesting a review.